### PR TITLE
 Physiological Model Integration with FHIR updated.

### PR DESCRIPTION
**Pull request by hopetes**
The implementation and the design are now consistent,
In each class, there is a description about what does the class do.

### CODE REVIEW

**Comment by basbergevoet** [comment]
@hopetes Hi Tesfay, where do I find the design? Is that Figure 5.2 in your thesis as it is currently stored on Teams?

**Comment by hopetes** [comment]
@basbergevoet Yes.
It is also in the repository.

**Comment by basbergevoet** [comment]
@hopetes I don't see a readme file on this pull request.......

I still would need explanation with the design picture, that I do not see in the Thesis yet. For instance:

- In general, the UML diagram talks about packages and components, while the implementation is all about classes that have relationships with each other. Why did you decide to create a UML Package diagram instead of a UML Class Diagram? Or at least a combination of the two? All following remarks are examples that explain this better.
- The classes defined in the Jupiter notebook are UML packages? I would expect a class is mapped on a UML class object. Is there a reason to deviate from this?
- Line 111: Apparently the User Interface class has a direct relation to the Data Management class. But I do not see that relation back in the design. Why not?
- Line 154: The PulseEngine and SEDataRequest class are imported from an external library (the Pulse Engine?). I assume these are classes as well? Why defined as UML component in the design diagram? 
- Why not just all UML classes, where the Jupiter notebook classes have a relation to the provided Pulse Engine classes? UML packages can then be used to cluster the Jupiter notebook classes and the Pulse Engine provided classes.
- The SimulationCore class interacts with the PulseEngine class to set actions, progress time, get results, etc. How is this relationship shown in the design diagram?
- Where are the SEAirwayObstruction class and FrequencyUnit class coming from? They are not in the design diagram. Or are they abstracted away? In that case, please explain in the text that explains the design diagram.
- The FHIR server is an external URL, whereas in the design diagram it is presented as a component in the DataManagement package (which is a class)....... Is that correct? Shouldn't that be modelled differently in UML?
- 
-

File: pulse-physiology_FHIR/ppe_fhir.ipynb [hunk 1]
```diff
@@ -125,18 +173,74 @@
     "            print(\"Unable to load initial state file\")\n",
     "        else:\n",
     "            self.results = self.pulse.pull_data()\n",
+    "        self.data_management.save_csv(self.results)\n",
+    "\n",
+    "    # Advances the time by 1 second and records the patient's state after each action\n",
+    "    def advance_time_and_record(self, start, seconds, action=None):\n",
```
Comment (PatrickBonne) [inline pulse-physiology_FHIR/ppe_fhir.ipynb:179]: The parameters start and seconds are not clear to me: what are the intended values?

Comment (PatrickBonne) [inline pulse-physiology_FHIR/ppe_fhir.ipynb:179]: if advance_time_and_record is meant to be a private function, please use the naming convention to start with an _
I was under the impression this was a public function on the SimulatorCore.

Comment (hopetes) [reply pulse-physiology_FHIR/ppe_fhir.ipynb:174]: This was to show the interval of time, but a little modification is made with the new commit.

Comment (hopetes) [reply pulse-physiology_FHIR/ppe_fhir.ipynb:174]: I made an appropriate access modifier to each function with the comming commit.

**Review by PatrickBonne — APPROVED** [review]

**Pull request merged**